substrate/client/network/src/protocol/notifications/tests/conformance.rs

```diff
@@ -0,0 +1,1067 @@
+							handle.close_substream(libp2p_peer.into()).await;
+
+							// After closing the substream set the timer to 6s to ensure the connection does
+							// not close due to the keep-alive mechanism of 5s.
+							timer = tokio::time::interval(std::time::Duration::from_secs(6));
+							timer.tick().await;
+						}
+					},
+					_ => {},
+				}
+			}
+
+			event = litep2p.next_event() => {
+				log::info!("[litep2p] event: {event:?}");
+			},
+
+			event = handle.next() => {
+				log::info!("[litep2p handle] event: {event:?}");
+
+				match event.unwrap() {
+					Litep2pNotificationEvent::ValidateSubstream { peer, handshake, .. } => {
+						assert_eq!(peer.to_bytes(), libp2p_peer.to_bytes());
+						assert_eq!(handshake, vec![1, 2, 3, 4]);
+
+						handle.send_validation_result(peer, Litep2pValidationResult::Accept);
+					},
+					Litep2pNotificationEvent::NotificationStreamOpened { peer, handshake, .. } => {
+						assert_eq!(peer.to_bytes(), libp2p_peer.to_bytes());
+						assert_eq!(handshake, vec![1, 2, 3, 4]);
+
+						handle.send_sync_notification(peer, vec![1, 1, 1, 1]).unwrap();
+						handle.send_async_notification(peer, vec![2, 2, 2, 2]).await.unwrap();
+					}
+					Litep2pNotificationEvent::NotificationReceived { peer, .. } => {
+						assert_eq!(peer.to_bytes(), libp2p_peer.to_bytes());
+
+					}
+					Litep2pNotificationEvent::NotificationStreamClosed { .. } => {
+					}
+					_ => {},
+				}
+			},
+		}
+	}
+}
+
+/// Raw litep2p closes the substream with a raw litep2p.
+/// In this case, there's no protocol controller from substrate that will reopen the substream.
+/// Therefore, since the substream is closed, the connection will be closed by the keep-alive.
+#[tokio::test]
+async fn litep2p_disconnects_litep2p_substream() {
+	let (mut litep2p_lhs, mut handle_lhs) = setup_litep2p().await;
+	let (mut litep2p_rhs, mut handle_rhs) = setup_litep2p().await;
+
+	let litep2p_lhs_peer = *litep2p_lhs.local_peer_id();
+	let litep2p_rhs_peer = *litep2p_rhs.local_peer_id();
+
+	let litep2p_address = litep2p_lhs.listen_addresses().into_iter().next().unwrap().clone();
+	litep2p_rhs.dial_address(litep2p_address).await.unwrap();
+
+	let mut num_closed = 0;
+	let mut lhs_opened = 0;
+	let mut lhs_closed = 0;
+	let mut rhs_opened = 0;
+	let mut rhs_closed = 0;
+
+	loop {
+		tokio::select! {
+			event = litep2p_lhs.next_event() => {
+				log::info!("[litep2p_lhs] event: {event:?}");
+
+				match event.unwrap() {
+					Litep2pEvent::ConnectionEstablished { .. } => {
+						handle_rhs.open_substream(litep2p_lhs_peer).await.unwrap();
+					}
+					Litep2pEvent::ConnectionClosed { .. } => {
+						num_closed += 1;
+						if num_closed == 2 {
+							break;
+						}
+					}
+					_ => {},
+				}
+			},
+
+			event = litep2p_rhs.next_event() => {
+				log::info!("[litep2p_rhs] event: {event:?}");
+
+				match event.unwrap() {
+					Litep2pEvent::ConnectionEstablished { .. } => {
+						handle_lhs.open_substream(litep2p_lhs_peer).await.unwrap();
```

Comment:
```suggestion
						handle_lhs.open_substream(litep2p_rhs_peer).await.unwrap();
```
But it should be enough for just one peer to open a substream to the other.